pennylane/templates/broadcast.py

```diff
@@ -64,6 +66,79 @@ def wires_all_to_all(wires):
 ###################
 
 
+def _preprocess(parameters, pattern, wires):
+    """Validate and pre-process inputs."""
```

Comment:
Add function argument documentation?

Reply:
True, that would be good. I was sloppy here...

Reply:
@josh146  you want me to add them?

Reply:
We could also be more precise about what preprocessing does in each case...

Reply:
Yeah, actually, it might be a good idea. Sorry I'm so used to this PR, I didn't even notice the docstrings were missing!